en/services/timesync_v1.md

```diff
@@ -0,0 +1,45 @@
+# Time Synchronization Protocol v1
+
+This protocol is used to estimate the time offsets between system clocks on two MAVLink systems, in a two-system MAVLink network.
+This is useful because it allows ....
+
+The protocol uses just one message [TIMESYNC](#TIMESYNC), which has two `int64_t` fields, `tc1` and `ts1`. 
+The message is sent out with a nanosecond timestamp in `ts1` (current unix timestamp * 1e9).
+This is mirrored back by the remote system, along with its own timestamp.
+The first system can then determine the round-trip time, and estimate the timestamp offset.
+This sequence is run multiple times to remove the transient channel and processor usage from the offset calculation.
+
+The protocol only works when there are only two systems on the MAVLink network.
+This is because [TIMESYNC](#TIMESYNC) is broadcast (cannot be addressed to a single node), so multiple components on the network may respond.
+A recipient has no way to reliably determine if the timestamp in `ts1` is the one it sent out, or is a reply to another system.
```

Comment:
```suggestion
[TIMESYNC](#TIMESYNC) is a broadcast message, so all components on the network will receive all `TIMESYNC` requests, and all components that support the protocol will send back `TIMESYNC` responses.
If there are multiple systems on the network, a system that is waiting on a `TIMESYNC` response cannot be _certain_ that a response to its request and not to that of another components.

A component can reduce the chance of clashes by appending its system ID to the bottom-most bits of the timestamp sent in the request, and only calculating the sync on responses that include this id.
This is approach is recommended, and should have no side effects, because the timestamp is simply mirrored by the pinged system.
```